### ISSUE TITLE
Cannot revoke a society admin a second time for same society

### ISSUE THREAD

**Issue by philosophicles** [camdram/camdram#367]
Society/venue admin status is stored in our `access` table. This permits multiple rows for one user and one society ("resource" = `rid`). Multiple rows are created if a user is made a society admin, then revoked, and then subsequently made an admin again. 

The `access` table has an identity PK, `id`, and conceptually also a compound unique key `rid`,`uid`,`creationdate`. (I suspect we'd run into interesting problems if someone became an admin, got revoked, and recreated all in one day - slightly poor table design, not really relevant to this issue.)

I try to revoke a society admin via the normal 'Edit the society's admins' page. If that admin has 2 rows in the access table as described above, the operation fails, showing our normal "Sorry, something's gone wrong" error page (i.e. is trapped by Symfony). I imagine something about the revocation code is expecting `rid` and `uid` to be unique. 

Here are the relevant lines from the Symfony log file (I've removed a few specifics that aren't publically available):

```
[2016-10-01 10:28:59] request.INFO: Matched route "revoke_society_admin". {"route_parameters":{"_controller":"Acts\\CamdramBundle\\Controller\\SocietyController::revokeAdminAction","_format":"html","identifier":"pembroke-players","_route":"revoke_society_admin"},"request_uri":"https://www.camdram.net/societies/pembroke-players/admin/revoke?uid=xxxx"} []
[2016-10-01 10:28:59] security.DEBUG: Read existing security token from the session. {"key":"_security_public"} []
[2016-10-01 10:28:59] security.DEBUG: User was reloaded from a user provider. {"username":"someone@somewhere.com","provider":"Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider"} []
[2016-10-01 10:28:59] request.CRITICAL: Uncaught PHP Exception Doctrine\ORM\NonUniqueResultException: "" at /var/www/camdram/20151223/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php line 781 {"exception":"[object] (Doctrine\\ORM\\NonUniqueResultException(code: 0): at /var/www/camdram/20151223/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php:781)"} []
[2016-10-01 10:28:59] security.DEBUG: Stored the security token in the session. {"key":"_security_public"} []
```

**Comment by GKFX** [comment]
Assigning myself as I'm poking around in here anyway for #191.

**Comment by philosophicles** [comment]
Recently raised - and fixed - issue #523 sounds extremely similar to this one. 

@CHTJonas are these actual duplicates? And @hoyes, since you fixed #523, do you think you've fixed this one with the same commit? (Sorry I haven't got time to test properly right now.)

**Comment by hoyes** [comment]
Ah... this does look like a duplicate - sorry if you'd already spent time on this @GKFX . I tested re-adding an admin to a venue earlier.

I went and removed the revoke* fields in acts_access and instead made it delete the row when access is revoked. A full history of who's done what when is still logged in prod.action.log if we ever need to know more and it makes the schema a little simpler. It also fixes some issues with foreign key constraints if the user who did the revoking is later deleted - it was hard to fix #498 fully without accidentally null-ing revokeid, thereby re-granting old access grants(!)
